source/renderer/app/stores/HardwareWalletsStore.ts

```diff
@@ -324,6 +312,72 @@ export default class HardwareWalletsStore extends Store {
       });
     }
   };
+
+  waitForLedgerDevices = async () => {
+    const device = await waitForLedgerDevicesChannel.request();
+    this.connectedHardwareWalletsDevices.set(device.path, device);
+    return device;
+  };
+
+  useCardanoAppInterval = (
```

Comment:
Is there any chance we could easily extract this function's logic to a separate module, to avoid growing this already gigantic file?

Reply:
If it requires a lot of effort, ignore this comment or leave a TODO. I don't want to block merging this